Sensus.Shared.iOS/Callbacks/iOSCallbackData.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright 2014 The Rector & Visitors of the University of Virginia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using UIKit;
+using Foundation;
+using UserNotifications;
+using Sensus.Shared.Callbacks;
+
+namespace Sensus.Shared.iOS.Callbacks
+{
+    public class iOSCallbackData : ICallbackData
+    {
+        #region Fields
+        private readonly NSDictionary _data;
+        #endregion
+
+        #region Constructors
+        public iOSCallbackData()
+        {
+            _data = new NSDictionary();
+        }
+
+        public iOSCallbackData(UNNotificationResponse data) : this(data?.Notification?.Request?.Content?.UserInfo)
+        { }
+
+        public iOSCallbackData(UILocalNotification local)
+        {
+            _data = local.UserInfo = new NSDictionary();

```

Comment:
Ahh yes. You are right. Should probably be something like this.

`_data = local.UserInfo = local.UserInfo ?? new NSDictionary();`